Python/thread_pthread.h

```diff
@@ -152,6 +152,31 @@ PyThread__init_thread(void)
  * Thread support.
  */
 
+/* The thread function argument passed to PyThread_start_new_thread has a
+ * different type compared to the thread function passed to the POSIX
+ * pthread_create function and the result of forcibly casting it to the latter
+ * type is undefined.  The pythread_fn struct and the pythread_helper_fn below
+ * help work around this in a conforming manner by wrapping around the thread
+ * function.
+ */
+typedef struct {
+    void (*func) (void *);
+    void *arg;
+} pythread_fn;
+
+static void *
+pythread_helper_fn(void *fn)
+{
+    pythread_fn *pfn = fn;
+
+    void (*func)(void *) = pfn->func;
+    void *arg = pfn->arg;
+
```

Comment:
nitpick: please move this empty line after the free, not before :-) to make it clear that the memory is freed when the function is called.